pkg/controller/localvolumeset/reconcile.go

```diff
@@ -0,0 +1,133 @@
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	LocalVolumeSetNameLabel      = "local.storage.openshift.io/localvolumeset-owner-name"
+	LocalVolumeSetNamespaceLabel = "local.storage.openshift.io/localvolumeset-owner-namespace"
+)
+
+// blank assignment to verify that ReconcileLocalVolumeSet implements reconcile.Reconciler
+var _ reconcile.Reconciler = &LocalVolumeSetReconciler{}
+
+// LocalVolumeSetReconciler reconciles a LocalVolumeSet object
+type LocalVolumeSetReconciler struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client    client.Client
+	scheme    *runtime.Scheme
+	reqLogger logr.Logger
+	lvSetMap  *lvSetMapStore
+}
+
+// Reconcile reads that state of the cluster for a LocalVolumeSet object and makes changes based on the state read
+// and what is in the LocalVolumeSet.Spec
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *LocalVolumeSetReconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	r.reqLogger = logf.Log.WithName(controllerName).WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	r.reqLogger.Info("Reconciling LocalVolumeSet")
+	// Fetch the LocalVolumeSet instance
+	instance := &localv1alpha1.LocalVolumeSet{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			r.lvSetMap.deregisterLocalVolumeSet(instance.Spec.StorageClassName, request.NamespacedName)
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		r.reqLogger.Error(err, "failed to get localvolumeset")
+		return reconcile.Result{}, err
+	}
+
+	// store a one to many association from storageClass to LocalVolumeSet
+	r.lvSetMap.registerLocalVolumeSet(instance.Spec.StorageClassName, request.NamespacedName)
+
+	// The diskmaker daemonset, local-staic-provisioner daemonset and configmap are created in pkg/daemon
+	// this way, there can be one daemonset for all LocalVolumeSets
+
+	err = r.syncStorageClass(instance)
+	if err != nil {
+		r.reqLogger.Error(err, "failed to sync storageclass")
+		return reconcile.Result{}, err
+	}
+
+	r.reqLogger.Info("updating status")
+	err = r.updateStatus(instance)
```

Comment:
so if I understand this correctly, we aren't setting Conditions field in status anywhere?

Reply:
fixed